### PR TITLE
Feat/tokenauth

### LINKED ISSUE #17: API: Create endpoint for registration of a new user
- [x] Provide functionality to register a new user
- [x] Provide REST endpoint path: `register`

### LINKED ISSUE #113: API: Create endpoint to login
- [x] Provide login functionality for an existing user
- [x] Provide REST endpoint path: `login`

### PR DESCRIPTION
**Pull request by Anafabula**
Implements a simple token authentication scheme including enpoints `/register`, `/login`, `/logout` and `/logoutall`.
`/register` and `/login` return a json object like this:
```json
{
	"expiry": "2022-06-22T01:37:05.620954Z",
	"token": "22aae158f7e89b67bdcadab912933f29fe83d81affb0963a15b99e31c4ecdb01"
}
```
The token needs to be sent in the [Authorization](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Authorization) header with `Token` as a Prefix like this:
```
Authorization: Token 22aae158f7e89b67bdcadab912933f29fe83d81affb0963a15b99e31c4ecdb01
```
Authorization can be disabled for an endpoint for testing in the views.py by annotating the function with `@permission_classes([AllowAny])` ([Example here, uncomment the line](https://github.com/naturerobots/HSOS-SEP-PlantMap-2022/blob/2d73e5f4360c42e71dcbb0cb546c3fd9a6d2dd05/django/restapi/views.py#L42)).

The logged-in user is accessible in the backend via `request.user`.

closes #17 
closes #113

### CODE REVIEW

File: django/restapi/urls.py [hunk 1]
```diff
@@ -7,8 +9,10 @@
 # Additionally, we include login URLs for the browsable API.
 
 urlpatterns = [
-    path('login', views.login),
-    path('register', views.register),
+    path('login', views.LoginView.as_view(), name='knox_login'),
+    path('logout', knox_views.LogoutView.as_view(), name='knox_logout'),
+    path('logoutall', knox_views.LogoutAllView.as_view(), name='knox_logoutall'),
```
Comment (jarkenau) [inline django/restapi/urls.py:14]: Do we really need an endpoint to log out all users? Or did I get something wrong?

File: django/restapi/views.py [hunk 2]
```diff
@@ -33,15 +39,32 @@
...
 # /login
-@api_view(['GET'])
-def login(request):
-    return JsonResponse({})
+class LoginView(KnoxLoginView):
+    permission_classes = (permissions.AllowAny,)
+
+    def post(self, request, format=None):
+        serializer = AuthTokenSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
```
Comment (jarkenau) [inline django/restapi/views.py:47]: Shouldn't the exception be caught and a 500 HTTP error code be returned?

File: django/restapi/urls.py (hunk 1 above)
Comment (Anafabula) [reply django/restapi/urls.py:14]: It logs out all sessions of the current user

File: django/restapi/views.py (hunk 2 above)
Comment (Anafabula) [reply django/restapi/views.py:47]: No, [500](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500) is for internal server problems. Currently it returns [401 Unauthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) and some detail like
```json
{
	"detail": "Authentication credentials were not provided."
}
```
or
```json
{
	"detail": "Invalid token."
}
```

**Review by jarkenau — APPROVED** [review]

**Review by pbrozi — APPROVED** [review]

**Pull request merged**